content/127labs/drone-k8s-helm/index.md

```diff
@@ -0,0 +1,54 @@
+---
+date: 2017-06-10T00:00:00+00:00
+title: Kubernetes Helm
+author: 127labs
+tags: [ kubernetes, helm, docker ]
+repo: 127labs/drone-k8s-helm
+logo: kubernetes_helm.svg
+image: 127labs/drone-k8s-helm
+---
+
+This Kubernetes plugin can be included into your pipelines to allow deployment with Helm.
+
+Example configuration:
+
+```yaml
+pipeline:
+  ship:
+    image: 127labs/drone-k8s-helm
+    chart: stable/dokuwiki
+    master: https://127.0.0.1:8001
+    token: super-long-token
+    release: wiki
+    skip_tls: true
+    clean_before_release: true
+    values:
+      dokuwikiEmail: $${DOKUWIKI_EMAIL}
+      dokuwikiPassword: $${DOKUWIKI_PASSWORD}
+    secrets: [dokuwiki_email, dokuwiki_password]
+```
+
+# Secret Reference
+
+Values are capable of accessing secrets by interpolating them with the `$${SECRET}` syntax. The plugin will
```

Comment:
The motivation here was to allow `PLUGIN_VALUES` env var to access secrets. We found out that by setting the `values` key (in 0.7) this way:
```yaml
values:
  dokuwikiEmail: $${DOKUWIKI_EMAIL}
  dokuwikiPassword: $${DOKUWIKI_PASSWORD}
```
is being translated into:
```
PLUGIN_VALUES={"dokuwikiEmail":"${DOKUWIKI_EMAIL}","dokuwikiPassword":"${DOKUWIKI_PASSWORD}"}
```
and if we have secrets:
```
DOKUWIKI_EMAIL=john@doe.com
DOKUWIKI_PASSWORD=password
```
being set for example. They will be available as env variable where `PLUGIN_VALUES` has access to.

I hope the explanation was clear and yes please, would really love to have your input in this.